tests/utils/GraphicalUnitTester.py

```diff
@@ -25,38 +26,59 @@ class SceneTester:
     -----------
     path_tests_medias_cache : : class:`str`
         Path to 'media' folder generated by manim. This folder contains cached data used by some tests.
-    path_tests_data : : class:`str`
+    path_control_data : : class:`str`
         Path to the data used for the tests (i.e the pre-rendered frames).
     scene : :class:`Scene`
         The scene tested
     """
 
-    def __init__(self, scene_object, module_tested, caching_needed=False):
+    def __init__(
+        self,
+        scene_object,
+        module_tested,
+        tmpdir,
+    ):
         # Disable the the logs, (--quiet is broken) TODO
         logging.disable(logging.CRITICAL)
+        tests_directory = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
         self.path_tests_medias_cache = os.path.join(
-            "tests", "tests_cache", module_tested
+            tmpdir,
+            "test_graphical_units",
+            "tests_cache",
+            module_tested,
+            scene_object.__name__,
+        )
+        self.path_control_data = os.path.join(
+            tests_directory, "control_data", "graphical_units_data", module_tested
         )
-        self.path_tests_data = os.path.join("tests", "tests_data", module_tested)
 
-        if caching_needed:
-            file_writer_config["text_dir"] = os.path.join(
-                self.path_tests_medias_cache, scene_object.__name__, "Text"
-            )
-            file_writer_config["tex_dir"] = os.path.join(
-                self.path_tests_medias_cache, scene_object.__name__, "Tex"
-            )
+        # IMPORTANT NOTE : The graphical units tests don't use for now any custom manim.cfg,
+        # since it is impossible to manually select a manim.cfg from a python file. (see issue #293)
+        file_writer_config["text_dir"] = os.path.join(
+            self.path_tests_medias_cache, "Text"
+        )
+        file_writer_config["tex_dir"] = os.path.join(
+            self.path_tests_medias_cache, "Tex"
+        )
 
         file_writer_config["skip_animations"] = True
-        file_writer_config["disable_caching"] = True
         file_writer_config["write_to_movie"] = False
+        file_writer_config["disable_caching"] = True
         config["pixel_height"] = 480
         config["pixel_width"] = 854
         config["frame_rate"] = 15
+
+        for dir_temp in [
```

Comment:
Question: is there any reason we are not using [tempfile](https://docs.python.org/3/library/tempfile.html)? If at all possible, we should definitely use it. There's also [pytest tempdir](https://docs.pytest.org/en/stable/tmpdir.html) which is perhaps more appropriate.

If it is possible to use either of these, we should leave that for a later PR.

Reply:
Oh wait I see I think the `tmpdir` argument is already a folder created by tempfile, no?

Reply:
It already used a `tempdir`, see l.45 of this file. 

>  think the tmpdir argument is already a folder created by tempfile, no?

Yep (but in fact, created by `pytest`)